### PR TITLE
Bugfix/mark ii device edit

### PR DESCRIPTION
**Pull request by chrisveilleux**
## Description
Add a ssh key validator endpoint as part of fixing an inability to save device edits on a Mark II

## How to test
Behave tests for the new endpoint have been added.  Run the account UI in conjunction with the changes in this PR to test the entire flow of adding an SSH Key

## Contributor license agreement signed?
CLA [Y]

### CODE REVIEW

File: api/account/tests/features/steps/pantacor_update.py [hunk 1]
```diff
@@ -27,7 +27,7 @@
 from selene.testing.device import add_device, add_pantacor_config
 
 
-@given("a device using Pantacor with manual updates enabled")
+@given("a device using Pantacor for continuous delivery")
```
Comment (krisgesling) [inline api/account/tests/features/steps/pantacor_update.py:30]: should the docstring for this function also be updated?

File: shared/selene/util/ssh/ssh.py [hunk 2]
```diff
@@ -101,3 +105,32 @@ def connect(self):
...
+    string. The length is packed as a big-endian unsigned integer.  The expected value
+    is 7 because the following string, 'ssh-rsa', is 7 bytes long.
+
+    :param public_key: key to validate
+    :return: boolean indicating if validation check passed.
+    """
+    is_valid = False
+    public_key_parts = public_key.split()
+    if len(public_key_parts) == 3:
+        key_type, key, _ = public_key.split()
```
Comment (krisgesling) [inline shared/selene/util/ssh/ssh.py:123]: I'm fairly sure the third component in an ssh-rsa key is optional. It's mostly used as an identifier of the creator, but officially is a `comment` so if:
`ssh-rsa ABC123 gez@local` was a valid key, then
`ssh-rsa ABC123` should also be valid

https://serverfault.com/questions/743548/what-significance-does-the-user-host-at-the-end-of-an-ssh-public-key-file-hold/743551

**Review by krisgesling — COMMENTED** [review]
Looks pretty good to me - haven't finished getting Selene setup yet though, so haven't done a manual test of the changes. 

The only potential issue is requiring the comment component of a public key. I don't believe that it's technically required so would confuse people if they are entering a key that works elsewhere but throws an error on Selene.

File: shared/selene/util/ssh/ssh.py (hunk 2 above)
Comment (chrisveilleux) [reply shared/selene/util/ssh/ssh.py]: nice catch.  fixed.

File: api/account/tests/features/steps/pantacor_update.py (hunk 1 above)
Comment (chrisveilleux) [reply api/account/tests/features/steps/pantacor_update.py:30]: I don't think the docstring is inaccurate.

Comment (krisgesling) [reply api/account/tests/features/steps/pantacor_update.py:30]: oh I took "continuous delivery" to mean that auto-updates were enabled

**Review by krisgesling — APPROVED** [review]
Great stuff!

**Pull request merged**